### PR TITLE
fix(GRA-1103): use separate method to refresh configs for gui

### PR DESCRIPTION
**Pull request by Aceix**
TESTING STEPS: check that CI builds pass

### CODE REVIEW

**Review by 0xdcarns — APPROVED** [review]

File: config/config.go [hunk 1]
```diff
@@ -131,22 +131,23 @@ func SetVersion(ver string) {
...
 
 // ReadNetclientConfig loads the host configuration into memory.
-func ReadNetclientConfig() {
+func ReadNetclientConfig() error {
 	lockfile := filepath.Join(os.TempDir(), ConfigLockfile)
 	file := GetNetclientPath() + "netclient.yml"
 	if err := Lock(lockfile); err != nil {
 		logger.Log(0, "unable to obtain lockfile for host config", err.Error())
-		return
+		return err
```
Comment (mattkasun) [inline config/config.go:139]: if you are going to return an err, you should probably not log the err here and let the caller do that if required.

File: config/config.go [hunk 2]
```diff
@@ -131,22 +131,23 @@ func SetVersion(ver string) {
...
 	}
 	defer Unlock(lockfile)
 	f, err := os.Open(file)
 	if err != nil {
 		logger.Log(0, "failed to open host config", err.Error())
-		return
+		return err
```
Comment (mattkasun) [inline config/config.go:145]: as per previous comment

File: config/config.go [hunk 3]
```diff
@@ -131,22 +131,23 @@ func SetVersion(ver string) {
...
 	}
 	if err := yaml.NewDecoder(f).Decode(&netclient); err != nil {
 		logger.Log(0, "failed to decode host config", err.Error())
 	}
+	return nil
```
Comment (mattkasun) [inline config/config.go:150]: as per previous comment

File: gui/exported.go [hunk 4]
```diff
@@ -36,8 +36,12 @@ func (app *App) GoJoinNetworkByToken(token string) (any, error) {
 
 // App.GoGetKnownNetworks returns all known network configs (node, server)
 func (app *App) GoGetKnownNetworks() ([]Network, error) {
-	configs := make([]Network, 0, 5)
+	// read fresh config from disk
+	if err := config.RefreshConfigs(); err != nil {
+		return nil, err
+	}
 
+	configs := make([]Network, 0, 5)
```
Comment (mattkasun) [inline gui/exported.go:44]: is there a reason to use make vice
config := []Network{} or
var config []Network

**Review by mattkasun — CHANGES_REQUESTED** [review]

File: gui/exported.go (hunk 4 above)
Comment (Aceix) [reply gui/exported.go:44]: so creation and memory allocation done in one step

Comment (mattkasun) [reply gui/exported.go:44]: in most cases (one network only) will be allocating more memory than required. In other cases(joining more than 5 networks), the append will result in reallocation.  Don't see the point, but not a huge deal.
For reference: google go style guide

Size hints and preallocation are important steps when combined with empirical analysis of the code and its integrations, to create performance-sensitive and resource-efficient code.

Most code does not need a size hint or preallocation, and can allow the runtime to grow the slice or map as necessary. It is acceptable to preallocate when the final size is known (e.g. when converting between a map and a slice) but this is not a readability requirement, and may not be worth the clutter in small cases.

Warning: Preallocating more memory than you need can waste memory in the fleet or even harm performance. When in doubt, see GoTip #3: Benchmarking Go Code and default to a zero initialization or a composite literal declaration.

**Review by mattkasun — APPROVED** [review]

File: gui/exported.go (hunk 4 above)
Comment (Aceix) [reply gui/exported.go:44]: right. got it

**Pull request merged**